src/InstrumentationEngine.Lib/FileLoggerSink.cpp

```diff
@@ -80,8 +80,25 @@ HRESULT CFileLoggerSink::Reset(_In_ LoggingFlags defaultFlags, _Out_ LoggingFlag
 
         // The file name is empty if (a) the last character in the path is a path separator
         // or (b) if no file name is given e.g. the file starts with a period.
-        // Generate a file name if the file name is empty and the ext is either empty or "."
-        if (tsFile.empty() && (0 == tsExtension.compare(_T(".")) || 0 == tsExtension.compare(_T(""))))
+        // Generate a file name if we're given a directory.
+        bool isDirectory = false;
+        if (tsFile.empty())
+        {
+            isDirectory = (0 == tsExtension.compare(_T(".")) || 0 == tsExtension.compare(_T("")));
+        }
+        else 
+        {
+            // Check if fileName is actually a directory, meaning the path separator was left off
+            DWORD fileFlags = GetFileAttributes(m_tsPathCandidate.c_str());
+            if ((fileFlags & INVALID_FILE_ATTRIBUTES) != INVALID_FILE_ATTRIBUTES && // path exists
```

Comment:
nit: this can just be `fileFlags != INVALID_FILE_ATTRIBUTES`, on `GetFileAttributes` failure it'll return that value exactly.